packages/components/organisms/f-offers/src/components/ModalCopiedVoucherCode.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+    <!-- Confirmation Modal, displayed when a user selects "Copy Code" on a promotional content card -->
+    <mega-modal
+        :is-open="isOpen"
+        is-narrow
+        :data-test-id="testIds.modal"
+        @close="closeModal"
+    >
+        <h3>{{ $t('offersInbox.voucherCodeModal.title') }}</h3>
+        <p v-html="$t('offersInbox.voucherCodeModal.body')" />
+        <a
+            v-if="hasLink"
+            :href="url"
+            :data-test-id="testIds.confirm"
+            class="o-btn o-btn--primary o-btn--block o-btn--rounded"
+        >
+            {{ $t('offersInbox.voucherCodeModal.confirmButton') }}
+        </a>
+        <button
+            type="button"
+            class="o-btn o-btn--tertiary o-btn--block"
+            :data-test-id="testIds.dismiss"
+            @click="closeModal"
+        >
+            {{ $t('offersInbox.voucherCodeModal.dismissButton') }}
+        </button>
+    </mega-modal>
+</template>
+
+<script>
+import MegaModal from '@justeat/f-mega-modal';
+import '@justeat/f-mega-modal/dist/f-mega-modal.css';
+
+export default {
+    components: {
+        MegaModal
+    },
+
+    props: {
+        isOpen: {
+            type: Boolean,
+            default: false
+        },
+        closeModal: {
```

Comment:
The typical pattern in Vue is to emit an event and trigger the function in the parent component, rather than passing function ala React.

Reply:
Agreed, thanks migrating the code from homeweb so will update accordingly.